tools/code-owners-parser/CodeOwnersParser/CodeOwnersFile.cs

```diff
@@ -59,13 +59,27 @@ public static List<CodeOwnerEntry> ParseContent(string fileContent)
             return entries;
         }
 
-        public static CodeOwnerEntry ParseAndFindOwnersForClosestMatch(string codeOwnersFilePathOrUrl, string targetPath)
+        public static CodeOwnerEntry ParseAndFindOwnersForClosestMatch(
+            string codeOwnersFilePathOrUrl,
+            string targetPath,
+            bool useNewFindOwnersForClosestMatchImpl = false)
         {
             var codeOwnerEntries = ParseFile(codeOwnersFilePathOrUrl);
-            return FindOwnersForClosestMatch(codeOwnerEntries, targetPath);
+            return FindOwnersForClosestMatch(codeOwnerEntries, targetPath, useNewFindOwnersForClosestMatchImpl);
         }
 
-        public static CodeOwnerEntry FindOwnersForClosestMatch(List<CodeOwnerEntry> codeOwnerEntries, string targetPath)
+        public static CodeOwnerEntry FindOwnersForClosestMatch(
+            List<CodeOwnerEntry> codeOwnerEntries,
+            string targetPath,
+            bool useNewFindOwnersForClosestMatchImpl = false)
+        {
+            return useNewFindOwnersForClosestMatchImpl
+                ? new MatchedCodeOwnerEntry(codeOwnerEntries, targetPath).Value
+                : FindOwnersForClosestMatchLegacyImpl(codeOwnerEntries, targetPath);
```

Comment:
Is there a reason you are leaving the old implementation? Or is that just temporary while you test out your new?

Reply:
OK I now understand your intention after reading your description. I guess I should have started there but I jumped directly into the code.

Reply:
@weshaggard I was also thinking to keep the old implementation at least for some time, in case the new implementation behavior leads to undesired effects. Then it will be easier to revert or reuse the old implementation. But probably after a week, or so, I would get rid of it.